rubocop/cop/cucumber/regex_step_name.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Cucumber
+      # Checks that Cucumber step definitions use Cucumber expressions
+      # instead of Regex. Note: this may not always be possible, and this
+      # cop is not smart enough to detect those cases.
```

Comment:
This means that the cop will have false failures in those cases, leading to false positive failing tests on PRs. That's not great. Perhaps that would be helped if the offense could be considered a low severity. Though I don't know if that will keep the test from failing while still showing up on GitHub, I'd have to investigate how reviewdog handles it.

But did find that [add_offense](https://www.rubydoc.info/gems/rubocop/RuboCop%2FCop%2FBase:add_offense) indeed supports a [severity level](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Severity).

Reply:
Here's what it looks like with `severity: :convention` https://github.com/brianjaustin/ao3-sandbox/pull/9/files

The action is still marked as failed, but at least the annotation looks a little bit less angry (the higher severities are either yellow or red with an :x:)

Reply:
As tested [here](https://github.com/brianjaustin/ao3-sandbox/pull/11/files) that unfortunately doesn't apply for commits from forks where reviewdog [degrades to annotations](https://github.com/reviewdog/reviewdog?tab=readme-ov-file#graceful-degradation-for-pull-requests-from-forked-repositories). Oh well, was worth the try.